web/fns.php

```diff
@@ -213,3 +213,32 @@ function db_connect() {
     $dbh->setAttribute(PDO::ATTR_ERRMODE, PDO::ERRMODE_EXCEPTION);
     return $dbh;
 }
+
+function createStreetManagerQuery() {
+
+    $non_primary_columns = ['promoter_organisation', 'location',
+        'works_location_type', 'proposed_start_date', 'proposed_end_date', 'work_category',
+        'work_status', 'traffic_management_type', 'permit_status', 'close_footway'];
+    $excluded_columns = $non_primary_columns;
+    foreach ($excluded_columns as $k => $v) {
+        $excluded_columns[$k] = "EXCLUDED." . $v;
+    }
+    $dbh = db_connect();
+    $query = $dbh->prepare("INSERT INTO streetmanager
+        (permit_reference_number, " . join(', ', $non_primary_columns) . ")
+        VALUES (?" . str_repeat(",?", count($non_primary_columns)) . ")
+        ON CONFLICT (permit_reference_number) DO UPDATE SET
+        (" . join(', ', $non_primary_columns) . ") = (" . join(', ', $excluded_columns) . ")
+    ");
+
+    return $query;
+}
+
+function insertIntoStreetManager($query, $object) {
+
+    $query->execute([
+		$object['permit_reference_number'], $object['project_name'], 'SRID=27700;' . $object['works_location_coordinates'],
+		$object['works_location_type'], $object['proposed_start_date'], $object['proposed_end_date'], 'work_cat',
+		'status', 'traffic_management', 'Permit', 'Footway'
```

Comment:
These have all become hard-coded, which will again break the existing street manager code